### DISCUSSION TITLE
Disable Save As

### DISCUSSION THREAD

**Discussion by billdenver** [sumatrapdfreader/sumatrapdf#3636] [General]
The problem we face when using SumatraPDF is that users can “Save As” creating a new file.  This causes multiple files and this gets worse as one user will annotate and save a copy and another user will annotate and save a different copy so we now have the original plus 2+++n different files with different notes, etc.  

Is it possible to open the current file with prior user’s notes if any, add notes and save the file, etc. but NOT have the ability to save to a different name or location.

**Comment by GitHubRulesOK** [comment]
To block editing you need to use an Adobe Reader and control Rights Managment which MuPDF does not need as its an annotation PDF reader with redaction for permanent deletion. Thus only needs to respect the file Open restriction not the Adobe options of locking user annotation or other internal piece editing of the PDF.

You could use restrictions.ini 
OR an earlier versions of SumatraPDF which showed Adobe annotations but did not itself use the Adobe annotation system!
OR simplest use Edge, but it too allows for editing annotations, unless managed.

Core problem is appending notes etc MANDATES save to a different file (even when using same filename, as the Widows OS does a fast shoe shuffle with naming) SumatraPDF as a personal viewer relies on a person all knowing what they want for file naming etc.

**Comment by billdenver** [comment]
We do not want to block editing at all.  We want to only allow edited documents to be saved back to the original – not create a new document.

**Comment by GitHubRulesOK** [comment]
When you close
 Adobe Acrobat or Edge or Foxit or PDFXedit
 and any change has been made to the rendering such as fill a field or correct a xref flaw or add annotation

In all cases you will be asked to save file and offered two choices (and ideally plus cancel)

The Default action in Acrobat Reader is block "save" until a change is made then offer "Save" as an alternative to "save as" then on accept  SAVE selection switch to "Save As" it may seem odd but that's the way Adobe do it.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a1d90b84-1e6a-43ef-a70f-c6674dd81e82)

Default Action.
A) save as new name at any location user requires this is the normal OS application functions "Save As"
This is reflected in SumatraPDF as only option is that default "Save As"
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/586177f8-616e-4809-9022-3ba49f775165)
Make annotation and SumatraPDF adds the option to Save to same File
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/faad9de3-7317-4da0-b9fa-c379417a02d2)
and comply with 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/eee88be1-1cb5-4543-80b3-94c70f1bb985)

So no issue with the enhancement to Adobe default method but to remove that default of "save as" would require much custom file handling.



B) (Same as Word Acrobat etc) save a temporary file somewhere that is not rights protected 
 since the original file will be temporarily "restricted" from any change by Windows File and Folder Handle controls
THEN close the current file and Windows can overwrite with the temporary file. This is what happens when we save as "same" file
It is NOT the same location in files index as the source file area is traditionally retained for recovery, but less so on SSD.

It is not easy to enforce B as sole action, since it is a potential for file system failure, when a file is on a read on media.
A series of file management tasks would be required to verify exact state of source file attributes and their folder rights too.
These are all managed by Windows system when a user choses save.

**Comment by billdenver** [comment]
Good explanation!  Perhaps we need to revisit our process.
